src/Microsoft.IdentityModel.Tokens/Encryption/SymmetricKeyWrapProvider.cs

```diff
@@ -150,6 +158,9 @@ protected virtual SymmetricAlgorithm GetSymmetricAlgorithm(SecurityKey key, stri
             if (key == null)
                 throw LogHelper.LogArgumentNullException(nameof(key));
 
+            if (!IsSupportedAlgorithm(key, algorithm))
```

Comment:
Not sure if this is required considering that ValidateKeySize() later on makes a similar check.

Reply:
@keegan-caruso @GeoK reviewing this points out a kink in our extensibility. IsSupportedAlgorithm is virtual, however ValidateKeySize is not and requires one of: Aes128KW, Aes128KeyWrap, Aes256KW, Aes256KeyWrap.

I opened https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/1569